src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -621,7 +625,16 @@ public List<UUID> getSourceDatasetIdsFromSnapshotRequest(
   public List<Dataset> getSourceDatasetsFromSnapshotRequest(
       SnapshotRequestModel snapshotRequestModel) {
     return snapshotRequestModel.getContents().stream()
-        .map(c -> datasetService.retrieveByName(c.getDatasetName()))
+        .map(
+            c ->
+                c.getMode() == SnapshotRequestContentsModel.ModeEnum.BYREQUESTID
+                    ? snapshotDao
+                        .retrieveSnapshot(
```

Comment:
Could use `retrieve()` here
```suggestion
                    ? retrieve(
```